elements/ethernet/etherrewrite.cc

```diff
@@ -71,12 +63,39 @@ Packet *
 EtherRewrite::pull(int)
 {
     if (Packet *p = input(0).pull())
-	return smaction(p);
+       return smaction(p);
     else
-	return 0;
+       return 0;
+}
+
+#if HAVE_BATCH
+PacketBatch *
+EtherRewrite::simple_action_batch(PacketBatch * batch)
+{
+    EXECUTE_FOR_EACH_PACKET(smaction, batch);
+    return batch;
 }
 
 void
+EtherRewrite::push_batch(int p, PacketBatch *batch)
+{
+    FOR_EACH_PACKET(batch, p) {
+        smaction(p);
```

Comment:
smaction may return another packet, so the batch would be broken.
You must use EXECUTE_FOR_EACH_PACKET(fnt,batch) that will repair the batch if the packet changes